firestore/distributed_counters.rb

```diff
@@ -0,0 +1,91 @@
+# Copyright 2020 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+
+def create_counter project_id:, num_shards:
+  # project_id = "Your Google Cloud Project ID"
+  # num_shards = "Number of shards for distributed counter"
+
+  firestore = Google::Cloud::Firestore.new project_id: project_id
```

Comment:
Can you include the client creation and the `require "google/cloud/firestore"` within the region tags